clock_control: L4/L0: Enable MSI as possible direct SYSCLK source on STM32 L0/L4

erwango opened the pull request:
MSI could be useful to provide an accurate clock when no external oscillator is available.

Add this option in STM32 clock_control driver.
Modify disco_l475_iot1 Kconfig.defconfig in order to propose 2 alternate default configurations.
MSI driven SYSCLK configuration is useful in BLE cases where accurate clock is needed.

erwango committed:
drivers: clock_control: Add MSI as possible clock source for stm32

Add MSI as potential SYSCLK direct source for L0 and L4.
It could be needed in some cases when an accurate clock
is required and no external clock is available.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ydamigos commented on boards/arm/disco_l475_iot1/disco_l475_iot1_defconfig:
```diff
@@ -17,16 +16,15 @@ CONFIG_GPIO=y
-CONFIG_CLOCK_STM32_PLL_R_DIVISOR=4
+# Alternative for BLE use cases: Direct HSI
+# CONFIG_CLOCK_STM32_PLL_SRC_HSI=y
+
```
MSI?

ydamigos commented on boards/arm/disco_l475_iot1/Kconfig.defconfig:
```diff
@@ -10,6 +10,33 @@ if BOARD_DISCO_L475_IOT1
+	default 2
+config CLOCK_STM32_PLL_R_DIVISOR
+	default 4
+endif # SYS_CLOCK_HW_CYCLES_PER_SEC=80000000
```
Please remove.

ydamigos commented on boards/arm/disco_l475_iot1/Kconfig.defconfig:
```diff
@@ -10,6 +10,33 @@ if BOARD_DISCO_L475_IOT1
+	default 4
+endif # SYS_CLOCK_HW_CYCLES_PER_SEC=80000000
+
+endif # CLOCK_STM32_PLL_SRC_HSI
```
Move this after
```
config SYS_CLOCK_HW_CYCLES_PER_SEC
	default 16000000
```

ydamigos reviewed (changes requested):
There is a mix with the `if` / `endif` in Kconfig.defconfig

erwango commented:
@ydamigos : thanks for review, I pushed too quickly. Should be better now

ydamigos commented on boards/arm/disco_l475_iot1/Kconfig.defconfig:
```diff
@@ -57,4 +86,6 @@ config VL53L0X_XSHUT_GPIO_PIN_NUM
 endif #VL53L0X
 
+
+
```
Could you remove this lines?

ydamigos commented on boards/arm/disco_l475_iot1/Kconfig.defconfig:
```diff
@@ -10,6 +10,35 @@ if BOARD_DISCO_L475_IOT1
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 16000000
+
+endif # CLOCK_STM32_SYSCLK_SRC_PLL
```
CLOCK_STM32_SYSCLK_SRC_MSI?

erwango committed:
boards: disco_l475_iot1: define 2 default clock configurations

In some BLE use cases, using MSI driven SYSCLK is required.
Rework board Kconfig.defconfig in order to enable use of 2
alternate default clock configurations.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@ydamigos : Done, thanks!

ydamigos reviewed (approved):
@erwango Thanks for the changes.
